### PR TITLE
Decorator to use drf_yasg when it's installed

### PR DESCRIPTION
**Pull request by jredrejo**
## Summary
Using _/api_explorer/_ with api endpoints handled by `rest_framework.views.APIView` is not too useful as it usually is not using a db model with a schema to be infered.
`drf_yasg` allows using a manual definition of the schema with the `swagger_auto_schema` but adding it to the code will break the production environment as this library is not installed.

This PR adds a new Python decorator to use `swagger_auto_schema` if it's installed and ignore it otherwise.
The decorator is documented with the params it needs and an example of usage. It supports GET,  PATCH, POST and PUT methods, but it does not allow all the [swagger_auto_schema](https://github.com/axnsan12/drf-yasg/blob/master/src/drf_yasg/utils.py#L33) supports, it's limited to the list of params, options as `required` or `operation_description` are not supported.

## Reviewer guidance
Using the new swagger_auto_schema_available decorator, decorate a get/put function of some api endpoint using  `rest_framework.views.APIView`  and check it appears correctly at the swagger /api_explorer/

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by rtibbles — COMMENTED** [review]
This seems fine, and looks like you've made the API a bit simpler than the default drf_yasg decorator, right?

Do you want to merge this in, or just wait for the feature branch that is using it to merge it in?

File: kolibri/utils/drf_utils.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from functools import wraps
```
Comment (rtibbles) [inline kolibri/utils/drf_utils.py]: Oh.. the only thing I can think of is whether we want to put it here or somewhere else? All our current DRF/API related machinery is in `kolibri/core` but I don't have a principled reason for it being in one place vs another.

**Comment by jredrejo** [comment]
> This seems fine, and looks like you've made the API a bit simpler than the default drf_yasg decorator, right?
> 
> Do you want to merge this in, or just wait for the feature branch that is using it to merge it in?

It can wait for the branch to be merged, I cherry-picked it there so it does not matter

File: kolibri/utils/drf_utils.py (hunk 1 above)
Comment (jredrejo) [reply kolibri/utils/drf_utils.py]: I don't have any preference, if you think moving the fileto `kolibri/core/utils` makes more sense, I'm happy with it

Comment (rtibbles) [reply kolibri/utils/drf_utils.py]: I think the subliminal distinction that I have made is that `kolibri/utils` is for general Python utils and `kolibri/core` functionality is for Django specific stuff. I don't know if the distinction holds true, but that's my interpretation of how we have roughly done things previously.

Comment (jredrejo) [reply kolibri/utils/drf_utils.py]: no problem, moved

**Comment by rtibbles** [comment]
Code already merged in with #9833!

**Pull request closed without merging**